packages/core/src/typeschema/types.ts

```diff
@@ -170,44 +278,71 @@ class StructureDefinitionParser {
       this.slicingContext.field.slices.push(this.slicingContext.current);
     }
     this.slicingContext.current = {
-      name: element.sliceName || '',
+      name: element.sliceName ?? '',
       fields: {},
-      min: element.min || 0,
+      min: element.min ?? 0,
       max: element.max === '*' ? Number.POSITIVE_INFINITY : Number.parseInt(element.max as string),
     };
   }
+
+  private parseFieldDefinition(ed: ElementDefinition): ElementValidator {
+    const max = parseCardinality(ed.max as string);
+    const baseMax = ed.base && ed.base.max ? parseCardinality(ed.base.max) : max;
+    const typedElementDef = { type: 'ElementDefinition', value: ed };
+    return {
+      min: ed.min ?? 0,
+      max: max,
+      isArray: baseMax > 1,
+      constraints: (ed.constraint ?? []).map((c) => ({
+        key: c.key ?? '',
```

Comment:
Just for safety, I'd prefer to ensure that the type is correct here so that nothing downstream ever encounters `undefined` instead of a string.  Nothing is using this structure yet, but when the code validating constraint expressions is added I'll make sure to cover the empty expression case